populate_db.sql

```diff
@@ -0,0 +1,216 @@
+
+-- Fetch CSV Data: (python script)
+-- TODO: Should parameterize the fetch csv data {{StartTime}}, {{EndTime}}
+
+-- select * from erc20.tokens
+
+-- select * from erc20."ERC20_evt_Transfer"
+-- WHERE '\x9008D19f58AAbD9eD0D60971565AA8510560ab41' in ("to", "from")
+-- and evt_block_time between '2022-03-01' and '2022-03-11'
+
+-- with
+-- token_times as (
+--     select 
+--         date_trunc('hour', evt_block_time) as hour,
+--         date_trunc('minute', evt_block_time) as minute, 
+--         contract_address 
+--     from erc20."ERC20_evt_Transfer"
+--     where '\x9008D19f58AAbD9eD0D60971565AA8510560ab41' in ("to", "from")
+--     and evt_block_time between '2022-03-01' and '2022-03-11'
+-- )
+
+-- prices.usd
+-- select pusd.* from prices.usd pusd
+-- inner join token_times tt
+-- on pusd.minute = tt.minute
+-- and pusd.contract_address = tt.contract_address
+
+-- prices_from_dex_data
+-- select p.* from prices.prices_from_dex_data p
+-- inner join token_times tt
+-- on p.hour = tt.hour
+-- and p.contract_address = tt.contract_address
+
+-- select * from prices.layer1_usd_eth
+-- where minute between '2022-03-01' and '2022-03-11'
+
+-- select * from gnosis_protocol_v2."trades"
+-- where block_time between '2022-03-01' and '2022-03-11'
+
+-- select * from gnosis_protocol_v2."batches"
+-- where block_time between '2022-03-01' and '2022-03-11'
+
+-- $ docker cp ./csvs/ postgres_container:/
+
+
+-- Create Schemas:
+CREATE SCHEMA erc20;
+CREATE SCHEMA prices;
+CREATE SCHEMA gnosis_protocol_v2;
+
+-- Shemas for each table were defined from dune more or less as follows:
+-- select 
+--     column_name, 
+--     data_type,
+--     is_nullable
+-- from information_schema.columns
+-- where table_schema = 'prices'
+-- and table_name = 'usd';
+
+-- Create Tables
+
+-- Tokens:
+CREATE TABLE erc20.tokens (
+   contract_address bytea UNIQUE,
+   symbol text,
+   decimals integer
+);
+CREATE INDEX IF NOT EXISTS tokens_contract_address_decimals_idx ON erc20.tokens USING btree (contract_address) INCLUDE (decimals);
+CREATE INDEX IF NOT EXISTS tokens_symbol_decimals_idx ON erc20.tokens USING btree (symbol) INCLUDE (decimals);
+
+CREATE TABLE erc20."ERC20_evt_Transfer" (
```

Comment:
At some point it would be great if the table creations could be done in a way that the information is copied over from dune. But sure, this is a future PR and not necessary for the MVP

Reply:
As nice as that would be, I don't know how it will happen. I have requested the table schemas from dune several times and I have essentially automated getting the info with this query:

```sql
select 
    column_name, 
    data_type,
    is_nullable
from information_schema.columns
where table_schema = 'prices'
and table_name = 'usd';
```

Seems easy enough.